Python/optimizer.c

```diff
@@ -182,6 +183,12 @@ _PyOptimizer_Optimize(
     if (err <= 0) {
         return err;
     }
+
+    if (++executors_created >= 10) {
+        executors_created = 0;
+        _Py_Executors_InvalidateOld(interp, 0);
```

Comment:
I haven't thought about it much, but it might make sense to do this *before* calling `opt->optimize`. That way peak memory usage is reduced, since we free *then* allocate, rather than allocate *then* free.